readthedocs/builds/forms.py

```diff
@@ -37,3 +39,77 @@ def save(self, commit=True):
         if obj.active and not obj.built and not obj.uploaded:
             trigger_build(project=obj.project, version=obj)
         return obj
+
+
+class RegexAutomationRuleForm(forms.ModelForm):
+
+    ALL_VERSIONS_REGEX = r'.*'
+    SEMVER_REGEX = r'^v?(\d+\.)(\d+\.)(\d)(-.+)?$'
+    MATCH_CHOICES = (
+        (ALL_VERSIONS_REGEX, 'All versions'),
+        (SEMVER_REGEX, 'Semver versions'),
+        (None, 'Custom'),
+    )
+
+    match = forms.ChoiceField(
+        label='Rule',
+        choices=MATCH_CHOICES,
+        initial=ALL_VERSIONS_REGEX,
+        required=False,
+    )
+
+    match_arg = forms.CharField(
+        label='Custom rule',
+        help_text=_('A Python regular expression'),
+        required=False,
+    )
+
+    class Meta:
+        model = RegexAutomationRule
+        fields = ['description', 'match', 'match_arg', 'version_type', 'action']
+
+    def __init__(self, *args, **kwargs):
+        self.project = kwargs.pop('project', None)
+        super().__init__(*args, **kwargs)
+
+        self.fields['version_type'].choices = [
```

Comment:
A comment saying that we are overriding this here to remove External and Unknown would be good.